### PR TITLE
usbredir: 0.9.0 -> 0.10.0

### PR DESCRIPTION
**Pull request by fabaff**
###### Motivation for this change
Update to latest upstream release 0.10.0

Change log: https://gitlab.freedesktop.org/spice/usbredir/-/blob/master/ChangeLog.md#usbredir-0100-27-may-2021

Upstream uses now `mason`.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

CC @offline

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129318` at fe4d0fd0 run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages marked as broken and skipped:</summary>
  <ul>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
  </ul>
</details>
<details>
  <summary>2 packages failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/0f73247e9f8e0ab489d8c62bea2bef47">aqemu</a></li>
    <li><a href="https://gist.github.com/5b429b67d2ba352e1b606faac8280b1b">cloud-init</a></li>
  </ul>
</details>
<details>
  <summary>22 packages skipped due to time constraints:</summary>
  <ul>
    <li>diffoscope</li>
    <li>libguestfs</li>
    <li>libvmi</li>
    <li>linuxPackages-libre.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualbox</li>
    <li>linuxPackages_xanmod.virtualbox</li>
    <li>multibootusb</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>qemu_full</li>
    <li>...</li>
  </ul>
</details>
<details>
  <summary>30 packages built successfully:</summary>
  <ul>
    <li>alpine-make-vm-image</li>
    <li>cloud-utils</li>
    <li>cot (python38Packages.cot)</li>
    <li>gnome.gnome-boxes</li>
    <li>lima</li>
    <li>linuxPackages.virtualbox (linuxPackages_5_10.virtualbox)</li>
    <li>linuxPackages_4_14.virtualbox</li>
    <li>linuxPackages_4_19.virtualbox</li>
    <li>linuxPackages_4_4.virtualbox</li>
    <li>linuxPackages_4_9.virtualbox</li>
    <li>linuxPackages_5_12.virtualbox</li>
    <li>linuxPackages_5_13.virtualbox (linuxPackages_latest.virtualbox)</li>
    <li>linuxPackages_5_4.virtualbox</li>
    <li>linuxPackages_hardened.virtualbox</li>
    <li>linuxPackages_latest_hardened.virtualbox</li>
    <li>linuxPackages_lqx.virtualbox</li>
    <li>linuxPackages_testing_bcachefs.virtualbox</li>
    <li>linuxPackages_zen.virtualbox</li>
    <li>open-watcom-bin</li>
    <li>out-of-tree</li>
    <li>python39Packages.cot</li>
    <li>qemu</li>
    <li>qemu-utils</li>
    <li>remmina</li>
    <li>solo5</li>
    <li>spice-gtk</li>
    <li>usbredir</li>
    <li>virt-manager</li>
    <li>virt-viewer</li>
    <li>virtualboxHardened</li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/meson-cmake.md">meson-cmake</a>

Meson uses CMake as a fallback dependency resolution method and it likely is not necessary here. The message about cmake not being found is purely informational.

Near pkgs/development/libraries/usbredir/default.nix:24:3:
```
   |
24 |   nativeBuildInputs = [
   |   ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/development/libraries/usbredir/default.nix [hunk 1]
```diff
@@ -1,26 +1,52 @@
...
+    sha256 = "1dz8jms9l6gg2hw0k6p1p1lnchc9mcgmskgvm5gbdvw3j7wrhdbz";
   };
 
-  NIX_CFLAGS_COMPILE = "-Wno-error";
+  nativeBuildInputs = [
+    cmake
+    glib
+    meson
+    ninja
+    pkg-config
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/usbredir/default.nix:29]: You are sure we need cmake, meson, ninja and pkg-config?

**Review by SuperSandro2000 — COMMENTED** [review]
diff LGTM

File: pkgs/development/libraries/usbredir/default.nix (hunk 1 above)
Comment (Ma27) [reply pkgs/development/libraries/usbredir/default.nix:29]: ninja/meson & pkg-config seem to be required, but I can compile without cmake as well.

File: pkgs/development/libraries/usbredir/default.nix [hunk 2]
```diff
@@ -1,26 +1,52 @@
...
+    domain = "gitlab.freedesktop.org";
+    owner = "spice";
+    repo = "usbredir";
+    rev = "${pname}-${version}";
+    sha256 = "1dz8jms9l6gg2hw0k6p1p1lnchc9mcgmskgvm5gbdvw3j7wrhdbz";
   };
 
-  NIX_CFLAGS_COMPILE = "-Wno-error";
+  nativeBuildInputs = [
+    cmake
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/usbredir/default.nix:25]:
```suggestion
```

File: pkgs/development/libraries/usbredir/default.nix [hunk 3]
```diff
@@ -1,26 +1,52 @@
...
+  ];
 
-  nativeBuildInputs = [ pkg-config ];
-  propagatedBuildInputs = [ libusb1 ];
+  propagatedBuildInputs = [
+    libusb1
+  ];
+
+  dontUseCmakeConfigure = true;
+
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/usbredir/default.nix:37]:
```suggestion
```

**Comment by Artturin** [comment]
@fabaff

File: pkgs/development/libraries/usbredir/default.nix (hunk 3 above)
Comment (fabaff) [reply pkgs/development/libraries/usbredir/default.nix]: Removed.

File: pkgs/development/libraries/usbredir/default.nix (hunk 2 above)
Comment (fabaff) [reply pkgs/development/libraries/usbredir/default.nix]: `cmake` is not needed. Removed.

**Comment by fabaff** [comment]
Result of `nixpkgs-review pr 129318` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>6 packages marked as broken and skipped:</summary>
  <ul>
    <li>aqemu</li>
    <li>linuxPackages-libre.virtualbox</li>
    <li>linuxPackages_hardkernel_4_14.virtualbox</li>
    <li>linuxPackages_hardkernel_latest.virtualbox</li>
    <li>linuxPackages_latest-libre.virtualbox</li>
    <li>multibootusb</li>
  </ul>
</details>
<details>
  <summary>49 packages built:</summary>
  <ul>
    <li>alpine-make-vm-image</li>
    <li>cloud-init</li>
    <li>cloud-utils</li>
    <li>cot (python39Packages.cot)</li>
    <li>diffoscope</li>
    <li>gnome.gnome-boxes</li>
    <li>libguestfs</li>
    <li>libvmi</li>
    <li>lima</li>
    <li>linuxPackages.virtualbox (linuxPackages_5_10.virtualbox)</li>
    <li>linuxPackages_4_14.virtualbox</li>
    <li>linuxPackages_4_19.virtualbox</li>
    <li>linuxPackages_4_4.virtualbox</li>
    <li>linuxPackages_4_9.virtualbox</li>
    <li>linuxPackages_5_12.virtualbox</li>
    <li>linuxPackages_5_13.virtualbox (linuxPackages_latest.virtualbox)</li>
    <li>linuxPackages_5_4.virtualbox</li>
    <li>linuxPackages_hardened.virtualbox</li>
    <li>linuxPackages_latest_hardened.virtualbox</li>
    <li>linuxPackages_lqx.virtualbox</li>
    <li>linuxPackages_testing_bcachefs.virtualbox</li>
    <li>linuxPackages_xanmod.virtualbox</li>
    <li>linuxPackages_zen.virtualbox</li>
    <li>open-watcom-bin</li>
    <li>out-of-tree</li>
    <li>python38Packages.cot</li>
    <li>python38Packages.guestfs</li>
    <li>python39Packages.guestfs</li>
    <li>qemu</li>
    <li>qemu-utils</li>
    <li>qemu_full</li>
    <li>qemu_kvm</li>
    <li>qemu_xen (qemu_xen_4_10)</li>
    <li>qemu_xen-light (qemu_xen_4_10-light)</li>
    <li>qtemu</li>
    <li>qubes-core-vchan-xen</li>
    <li>remmina</li>
    <li>solo5</li>
    <li>spice-gtk</li>
    <li>usbredir</li>
    <li>vagrant</li>
    <li>virt-manager</li>
    <li>virt-manager-qt</li>
    <li>virt-viewer</li>
    <li>virtualboxHardened</li>
    <li>virtualboxHeadless</li>
    <li>virtualboxWithExtpack</li>
    <li>xen (xenPackages.xen-vanilla ,xenPackages.xen_4_10-vanilla ,xen_4_10)</li>
    <li>xen-slim (xenPackages.xen-slim ,xenPackages.xen_4_10-slim ,xen_4_10-slim)</li>
  </ul>
</details>

**Pull request merged**